app/decorators/telegram_message.rb

```diff
@@ -82,4 +87,20 @@ def initialize_photos(telegram_message)
     photo.image.attach(io: URI.open(remote_file_location), filename: File.basename(remote_file_location.path))
     [photo]
   end
+
+  def initialize_voice(telegram_message)
+    return nil unless telegram_message[:voice]
+
+    voice = Voice.new
+    bot_token = "bot#{Telegram.bots[Rails.configuration.bot_id].token}"
+    telegram_file = telegram_message[:voice]
+    file_id = telegram_file[:file_id]
+    uri = URI("https://api.telegram.org/#{bot_token}/getFile")
+    uri.query = URI.encode_www_form({ file_id: file_id })
+    response = JSON.parse(URI.open(uri).read)
+    file_path = response.dig('result', 'file_path')
+    remote_file_location = URI("https://api.telegram.org/file/#{bot_token}/#{file_path}")
+    voice.attachment.attach(io: URI.open(remote_file_location), filename: File.basename(remote_file_location.path))
+    voice
```

Comment:
```suggestion
    retrieve_message_type_and_attach(message_type)
```

I think this code is so similar to what we do for photos that we could DRY up the code a bit by extracting it to its own method...